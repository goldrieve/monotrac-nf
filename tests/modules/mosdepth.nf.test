nextflow_process {

    name "Test Process MOSDEPTH"
    script "main.nf"
    process "MOSDEPTH"

    test("Should run mosdepth") {

        when {
            params {
            }
            process {
                """
                input[0] = ["SVP_test", file("${projectDir}/tests/data/SVP_test.bam.bai"), file("${projectDir}/tests/data/SVP_test.bam")] 
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
