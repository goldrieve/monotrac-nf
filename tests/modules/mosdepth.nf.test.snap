{
    "Should run mosdepth": {
        "content": [
            {
                "0": [
                    "SVP_test.mosdepth.global.dist.txt:md5,ab6c3845e8ab162ee66cc8845f555a51"
                ],
                "1": [
                    [
                        "SVP_test",
                        "SVP_test.mosdepth.global.dist.txt:md5,ab6c3845e8ab162ee66cc8845f555a51"
                    ]
                ],
                "2": [
                    "SVP_test.mosdepth.summary.txt:md5,f7154a0bc2c281a51719b24a66c507be"
                ],
                "global": [
                    "SVP_test.mosdepth.global.dist.txt:md5,ab6c3845e8ab162ee66cc8845f555a51"
                ],
                "sample_global": [
                    [
                        "SVP_test",
                        "SVP_test.mosdepth.global.dist.txt:md5,ab6c3845e8ab162ee66cc8845f555a51"
                    ]
                ],
                "summary": [
                    "SVP_test.mosdepth.summary.txt:md5,f7154a0bc2c281a51719b24a66c507be"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.7"
        },
        "timestamp": "2025-10-06T10:39:50.877193316"
    }
}